src/bridge/cockpitdbusjson.c

```diff
@@ -2917,24 +2956,48 @@ cockpit_dbus_json_prepare (CockpitChannel *channel)
       else
         self->logname = bus;
 
-      /* Ready when the bus connection is available */
-      if (self->bus_type == G_BUS_TYPE_NONE)
+      if (self->bus_type != G_BUS_TYPE_NONE)
         {
-          GDBusConnectionFlags flags = G_DBUS_CONNECTION_FLAGS_AUTHENTICATION_CLIENT;
-          if (self->default_name)
-            flags = flags | G_DBUS_CONNECTION_FLAGS_MESSAGE_BUS_CONNECTION;
+          bus_address = g_dbus_address_get_for_bus_sync (self->bus_type, self->cancellable, NULL);
+          address = bus_address;
```

Comment:
I have a feeling this might get flagged by coverity as potentially overriding the const address. Even though address is supposed to only be used with G_BUS_TYPE_NONE. Do we need an additional check above to make sure that is enforced?